Assets/Scripts/EnemySystem/ScriptableObjects/EnemyAttackStrategies/BabeAttackStrategy.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Player;
+using UnityEngine;
+
+namespace EnemySystem.ScriptableObjects.EnemyAttackStrategies
+{
+    [CreateAssetMenu(fileName = "BabeAttackStrategy", 
+        menuName = "ScriptableObjects/EnemyAttack/Babe", order = 3)]
+    public class BabeAttackStrategy : EnemyAttackStrategy
+    {
+        private float rangeOfMeleeAttack = 2f;
+        private float waitTime = 2;
+
+        public override void Attack(GameObject player, GameObject emu, int damageGiven)
+        {
+            if (player.transform != null &&
+                Vector2.Distance(emu.transform.position, player.transform.position) < rangeOfMeleeAttack)
+            {
+                var playerController = player.GetComponent<Controller>();
```

Comment:
Instead of fetching the player controller every time you attack, it would be more efficient to fetch the player controller once in a `Start()` method

Reply:
You can't use Start() in Strategies :D

Reply:
oh gross